subprojects/execution/src/test/groovy/org/gradle/internal/execution/steps/ValidateStepTest.groovy

```diff
@@ -76,10 +76,9 @@ class ValidateStepTest extends StepSpec<AfterPreviousExecutionContext> {
 
         then:
         def ex = thrown WorkValidationException
-        ex.message == "Some problems were found with the configuration of job ':test' (types 'ValidateStepTest.JobType', 'ValidateStepTest.SecondaryJobType')."
-        ex.causes.size() == 2
-        ex.causes[0].message == "Type '$Object.simpleName': Validation error #1."
-        ex.causes[1].message == "Type '$Object.simpleName': Validation error #2."
+        normaliseLineSeparators(ex.message) == """Some problems were found with the configuration of job ':test' (types 'ValidateStepTest.JobType', 'ValidateStepTest.SecondaryJobType').
+  - Type '$Object.simpleName': Validation error #1.
```

Comment:
I think we are losing some indentation here: We currently have:
```
Validation failed for task ':docs:userguideMultiPage', disabling optimizations:
   - Type 'AsciidoctorTask': non-property method 'asGemPath()' should not be annotated with: @Optional, @InputDirectory.
```

And IIUC, the new code would not indent the second line, right?

Reply:
That's a generic thing with tree formatter : it only creates child nodes if there are more than one. Do you feel strongly we would need to indent _even with a single node_ here? It's not what we do with other problems, typically in dependency management.

Reply:
My understanding is that with multiple nodes there is no indenting and with one node there isn't even a new line. Is that correct? I was wondering if then indenting could be changed with treeformatter, though I don't feel strongly enough to do something about it if it is not possible, yet.

Reply:
There is indenting for a single node. There's no new line if there's a single child. E.g:

```
This is a message: with a single error
```

and:

```
This is a message:
   - with a first error
   - and a second one
 ```

Reply:
Then all is good. Looks like I read the tests wrong :(